src/dune_rules/dune_file.ml

```diff
@@ -2245,6 +2245,27 @@ module Generate_sites_module = struct
        { loc; module_; sourceroot; relocatable; sites; plugins })
 end
 
+module Melange_stanza = struct
+  type t =
+    { loc : Loc.t
+    ; target : string
+    ; spec : Melange.Spec.t
+    ; libraries : Lib_dep.t list
+    }
+
+  let decode =
+    fields
+      (let+ loc = loc
+       and+ target = field "target" string
+       and+ spec =
+         field "spec"
```

Comment:
can we call this `module_system` perhaps?

Reply:
much better naming 👍  renamed in https://github.com/ocaml/dune/pull/6349/commits/15cbcd0979b2ed55e6865311925eedb69b369eb8.